Gulpfile.js

```diff
@@ -1,83 +1,99 @@
-    'node_modules/systemjs/dist/system-polyfills.js',
-    'node_modules/systemjs/dist/system.src.js',
-    'node_modules/zone.js/dist/**/*.+(js|js.map)',
-    'node_modules/reflect-metadata/**/*.+(ts|js|js.map)',
-    'node_modules/@angular/**/*.+(js|js.map)',
-    'node_modules/es6-shim/es6-shim.min.js',
-    'node_modules/jquery/dist/jquery.min.js',
-    'node_modules/bootstrap/dist/js/bootstrap.min.js',
-    'node_modules/bootstrap/dist/css/bootstrap.min.css',
-    'node_modules/bootstrap/dist/fonts/**/*.*',
-    'node_modules/font-awesome/css/font-awesome.min.css',
-    'node_modules/core-js/client/shim.min.js'
-];
+    vendors    = ['node_modules/rxjs/**/*.+(js|js.map)',
+                  'node_modules/systemjs/dist/system-polyfills.js',
+                  'node_modules/systemjs/dist/system.src.js',
+                  'node_modules/zone.js/dist/**/*.+(js|js.map)',
+                  'node_modules/reflect-metadata/**/*.+(ts|js|js.map)',
+                  'node_modules/@angular/**/*.+(js|js.map)',
+                  'node_modules/es6-shim/es6-shim.min.js',
+                  'node_modules/jquery/dist/jquery.min.js',
+                  'node_modules/bootstrap/dist/js/bootstrap.min.js',
+                  'node_modules/bootstrap/dist/css/bootstrap.min.css',
+                  'node_modules/bootstrap/dist/fonts/**/*.*',
+                  'node_modules/font-awesome/css/font-awesome.min.css',
+                  'node_modules/core-js/client/shim.min.js'],
+                 
+    injects    = ['dist/vendor/reflect-metadata/Reflect.js',
+                  'dist/vendor/core-js/client/shim.min.js',
+                  'dist/vendor/systemjs/dist/system.src.js',
+                  'dist/vendor/zone.js/dist/zone.js',
+                  'dist/vendor/jquery/dist/jquery.min.js',
+                  'dist/vendor/bootstrap/dist/js/bootstrap.min.js',
+                  'dist/vendor/bootstrap/dist/css/bootstrap.min.css',
+                  'dist/vendor/font-awesome/css/font-awesome.min.css'];
 
-var injects = [
-    paths.dest + '/vendor/reflect-metadata/Reflect.js',
-    paths.dest + '/vendor/core-js/client/shim.min.js',
-    paths.dest + '/vendor/systemjs/dist/system.src.js',
-    paths.dest + '/vendor/zone.js/dist/zone.js',
-    paths.dest + '/vendor/jquery/dist/jquery.min.js',
-    paths.dest + '/vendor/bootstrap/dist/js/bootstrap.min.js',
-    paths.dest + '/vendor/bootstrap/dist/css/bootstrap.min.css',
-    paths.dest + '/vendor/font-awesome/css/font-awesome.min.css'
-];
+////////////////////////////////////////////////////////////////////////////////////////////////////
 
-gulp.task('compile', function () {
-    var tsProject = typescript.createProject('src/tsconfig.json');
-    return gulp.src(paths.typescript)
-        .pipe(sourcemaps.init())
-        .pipe(typescript(tsProject))
-        .pipe(uglify())
-        .pipe(sourcemaps.write('.'))
-        .pipe(gulp.dest(paths.dest));
+gulp.task('clean', function () {
+    return del(['dist/**']);
```

Comment:
consider using paths.dest for consistency and so you only need to update the variable to change it.